src/ledger/LedgerTxn.cpp

```diff
@@ -26,7 +26,89 @@
 
 namespace stellar
 {
-
+namespace
+{
+void
+enforcePreLoadInvariants(LedgerKey const& key, uint32_t ledgerVersion)
+{
+    switch (key.type())
```

Comment:
Nit: Since `TRUSTLINE` is the only type here that needs this check, probably cleaner to use and if statement instead of switch.

Reply:
Agreed, I wrote the switch before I realized how little invariants we are actually enforcing 😅